model/websocket_message.go

```diff
@@ -33,20 +33,26 @@ type WebSocketMessage interface {
 	IsValid() bool
 }
 
+type WebsocketBroadcast struct {
+	OmitUsers map[string]bool `json:"-"`          // broadcast is omitted for users listed here
+	UserId    string          `json:"user_id"`    // broadcast only occurs for this user
+	ChannelId string          `json:"channel_id"` // broadcast only occurs for users in this channel
+	TeamId    string          `json:"team_id"`    // broadcast only occurs for users in this team
+}
+
 type WebSocketEvent struct {
-	TeamId    string                 `json:"team_id"`
-	ChannelId string                 `json:"channel_id"`
-	UserId    string                 `json:"user_id"`
 	Event     string                 `json:"event"`
 	Data      map[string]interface{} `json:"data"`
+	Broadcast *WebsocketBroadcast    `json:"broadcast"`
```

Comment:
Why a new struct and not just add the needed fields? Is it just for readability or is actually needed separately somewhere?